lib/compat/wordpress-6.5/interactivity-api/interactivity-api.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Interactivity API: Functions and hooks
+ *
+ * @package WordPress
+ * @subpackage Interactivity API
+ */
+
+if ( ! function_exists( 'wp_interactivity_process_directives_of_interactive_blocks' ) ) {
+	/**
+	 * Processes the directives on the rendered HTML of the interactive blocks.
+	 *
+	 * It only processes one root interactive block at a time because the rendered
+	 * HTML of that block contains the rendered HTML of all its inner blocks,
+	 * including any interactive block. It does so by ignoring all the interactive
+	 * inner blocks until the root interactive block is processed.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param array $parsed_block The parsed block.
+	 * @return array The same parsed block.
+	 */
+	function wp_interactivity_process_directives_of_interactive_blocks( $parsed_block ) {
+		static $root_interactive_block = null;
+
+		if ( null === $root_interactive_block ) {
+			$block_name = $parsed_block['blockName'];
+			$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
+
+			if ( isset( $block_name ) && isset( $block_type->supports['interactivity'] ) && $block_type->supports['interactivity'] ) {
+				$root_interactive_block = array( $block_name, md5( serialize( $parsed_block ) ) );
+
+				$process_interactive_blocks = static function ( $content, $parsed_block ) use ( &$root_interactive_block ) {
+					list($root_block_name, $root_block_md5) = $root_interactive_block;
+					if ( $root_block_name === $parsed_block['blockName'] && md5( serialize( $parsed_block ) ) === $root_block_md5 ) {
+						$root_interactive_block = null;
+						$content                = wp_interactivity_process_directives( $content );
+					}
+					return $content;
+				};
+
+				add_filter( 'render_block', $process_interactive_blocks, 10, 2 );
```

Comment:
Is this line creating a queue of filters with the same callback, per root interactive block in the post_content?

Maybe we can add a `remove_filter` in line 38, inside the conditional.

Reply:
Good point. Added in https://github.com/WordPress/gutenberg/pull/58066/commits/56f37470c58984aac9f29a7aa5ba0fd8c2c641d3.